manager-dashboard/app/views/Login/index.tsx

```diff
@@ -0,0 +1,191 @@
+import React from 'react';
+import { _cs } from '@togglecorp/fujs';
+import {
+    signInWithEmailAndPassword,
+    getAuth,
+    AuthError,
+    AuthErrorCodes,
+} from 'firebase/auth';
+import {
+    ObjectSchema,
+    requiredStringCondition,
+    useForm,
+    getErrorObject,
+    createSubmitHandler,
+    internal,
+} from '@togglecorp/toggle-form';
+
+import TextInput from '#components/TextInput';
+import Button from '#components/Button';
+
+import useMountedRef from '#hooks/useMountedRef';
+
+import mapSwipeLogo from '#resources/images/mapswipe-logo.svg';
+
+import styles from './styles.css';
+
+interface LoginFormFields {
+    email?: string | undefined;
+    password?: string | undefined;
+}
+
+type LoginFormSchema = ObjectSchema<LoginFormFields>;
+type LoginFormSchemaFields = ReturnType<LoginFormSchema['fields']>
+const loginFormSchema: LoginFormSchema = {
+    fields: (): LoginFormSchemaFields => ({
+        email: [requiredStringCondition],
+        password: [requiredStringCondition],
+    }),
+};
+
+const defaultLoginFormValue: LoginFormFields = {};
+
+interface Props {
+    className?: string;
+}
+
+function Login(props: Props) {
+    const {
+        className,
+    } = props;
+
+    const mountedRef = useMountedRef();
+
+    const {
+        setFieldValue,
+        error: formError,
+        value,
+        validate,
+        setError,
+    } = useForm(loginFormSchema, defaultLoginFormValue);
+    const error = getErrorObject(formError);
+
+    const [pending, setPending] = React.useState(false);
+
+    const handleFormSubmission = React.useCallback((finalValues: LoginFormFields) => {
+        async function login() {
+            if (!finalValues || !finalValues.email || !finalValues.password) {
+                // eslint-disable-next-line no-console
+                console.error('Email or password is not defined');
+                return;
+            }
+
+            try {
+                setPending(true);
+
+                const auth = getAuth();
+                await signInWithEmailAndPassword(
+                    auth,
+                    finalValues.email as string,
+                    finalValues.password as string,
+                );
+                // NOTE: we will udpate the current user on <Init />
+                if (!mountedRef.current) {
+                    return;
+                }
+                setErrorMessage(undefined);
+                setPending(false);
+            } catch (submissionError) {
```

Comment:
just so that I can better understand this part: Where do you define that only project managers can login through the manager dashboard? Normal mapswipe users will not be able to see/login for the manager dashboard, only user account with project manager credentials in firebase will be able to do this. Is this handled here or in another file?

Reply:
It's defined on a separate file.

https://github.com/mapswipe/python-mapswipe-workers/blob/0d013617a60b747643ae21bd9b42fbff295fd687/manager-dashboard/app/Base/components/Init/index.tsx#L67-L77